docs/src/topics/sparse_matrix.md

```diff
@@ -0,0 +1,165 @@
+    but this is not true in general since nodes and DoFs can be numbered independently (and
+    in fact are numbered independently in Ferrite).
+
+    ```
+    1 ----- 2 ----- 3 ----- 4
+    ```
+
+    Assuming we use linear Lagrange interpolation (the "hat functions") this will give the
+    following connections according to the weak form:
+     - Trial function 1 couples with test functions 1 and 2 (entries `(1, 1)` and `(1, 2)`
+       included in the sparsity pattern)
+     - Trial function 2 couples with test functions 1, 2, and 3 (entries `(2, 1)`, `(2, 2)`,
+       and `(2, 3)` included in the sparsity pattern)
+     - Trial function 3 couples with test functions 2, 3, and 4 (entries `(3, 2)`, `(3, 3)`,
+       and `(3, 4)` included in the sparsity pattern)
+     - Trial function 4 couples with test functions 3 and 4 (entries `(4, 3)` and `(4, 4)`
+       included in the sparsity pattern)
+
+    The resulting sparsity pattern would look like this:
+
+    ```
+    4×4 SparseArrays.SparseMatrixCSC{Float64, Int64} with 10 stored entries:
+     0.0  0.0   ⋅    ⋅
+     0.0  0.0  0.0   ⋅
+      ⋅   0.0  0.0  0.0
+      ⋅    ⋅   0.0  0.0
+    ```
+
+    Moreover, if the problem is solved with periodic boundary conditions, for example by
+    constraining the value on the right side to the value on the left side, there will be
+    additional couplings. In the example above, this means that DoF 4 should be equal to DoF
+    1. Since DoF 4 is constrained it has to be eliminated from the system. Existing entries
+    that include DoF 4 are `(3, 4)`, `(4, 3)`, and `(4, 4)`. Given the simple constraint in
+    this case we can simply replace DoF 4 with DoF 1 in these entries and we end up with
+    entries `(3, 1)`, `(1, 3)`, and `(1, 1)`. This results in two new entries: `(3, 1)` and
+    `(1, 3)` (entry `(1, 1)` is already included).
+
+## Creating sparsity patterns
+
+Creating a sparsity pattern can be quite expensive if not done properly and therefore
+Ferrite provides efficient methods and datastructures for this. In general the sparsity
+pattern is not known in advance and has to be created incrementally. To make this
+incremental construction efficient it is necessary to use a dynamic data structure which
+allow for fast insertions.
+
+The sparsity pattern also serves as a "matrix builder". When all entries are inserted into
+the sparsity pattern the dynamic data structure is typically converted, or "compressed",
+into a sparse matrix format such as e.g. the [*compressed sparse row
+(CSR)*](https://en.wikipedia.org/wiki/Sparse_matrix#Compressed_sparse_row_(CSR,_CRS_or_Yale_format))
+format or the [*compressed sparse column
+(CSC)*](https://en.wikipedia.org/wiki/Sparse_matrix#Compressed_sparse_column_(CSC_or_CCS))
+format, where the latter is the default sparse matrix type implemented in the SparseArrays
+standard library. These matrix formats allow for fast linear algebra operations, such as
```

Comment:
Maybe we should just add a link to make sure this is clear for new users which are not too familiar with the ecosystem.
```suggestion
format, where the latter is the default sparse matrix type implemented in the [SparseArrays
standard library](https://sparsearrays.juliasparse.org/dev/). These matrix formats allow for fast linear algebra operations, such as
```